controllers/openstackcluster_controller.go

```diff
@@ -196,24 +196,34 @@ func deleteBastion(log logr.Logger, osProviderClient *gophercloud.ProviderClient
 		return err
 	}
 
-	instance, err := computeService.GetInstanceByName(openStackCluster, fmt.Sprintf("%s-bastion", cluster.Name))
+	instanceStatus, err := computeService.GetInstanceByName(openStackCluster, fmt.Sprintf("%s-bastion", cluster.Name))
```

Comment:
not sure we need rename to GetInstanceStatusByName?

Reply:
I'm inclined to leave it for now because it's already getting a bit long, but it's easy to change if we decide it's confusing. I think it should be ok, though, because now everything returns InstanceStatus.

Reply:
+1 for rename it to `GetInstanceStatusByName`